pkg/xds/server/components.go

```diff
@@ -1,405 +1,22 @@
-					return err
-				}
-
-				// create creates a map of selectors to match other dataplanes reachable via given routes
-				destinations := xds_topology.BuildDestinationMap(dataplane, routes)
-
-				// resolve all endpoints that match given selectors
-				outbound := xds_topology.BuildEndpointMap(
-					meshRes, rt.Config().Multizone.Remote.Zone,
-					dataplanes.Items, externalServices.Items, rt.DataSourceLoader())
-
-				healthChecks, err := xds_topology.GetHealthChecks(ctx, dataplane, destinations, rt.ReadOnlyResourceManager())
-				if err != nil {
-					return err
-				}
-
-				circuitBreakers, err := xds_topology.GetCircuitBreakers(ctx, dataplane, destinations, rt.ReadOnlyResourceManager())
-				if err != nil {
-					return err
-				}
-
-				trafficTrace, err := xds_topology.GetTrafficTrace(ctx, dataplane, rt.ReadOnlyResourceManager())
-				if err != nil {
-					return err
-				}
-				var tracingBackend *mesh_proto.TracingBackend
-				if trafficTrace != nil {
-					tracingBackend = meshRes.GetTracingBackend(trafficTrace.Spec.GetConf().GetBackend())
-				}
-
-				matchedPermissions, err := permissionsMatcher.Match(ctx, dataplane, meshRes)
-				if err != nil {
-					return err
-				}
-
-				matchedLogs, err := logsMatcher.Match(ctx, dataplane)
-				if err != nil {
-					return err
-				}
-
-				faultInjection, err := faultInjectionMatcher.Match(ctx, dataplane, meshRes)
-				if err != nil {
-					return err
-				}
-
-				proxy := xds.Proxy{
-					Id:                 proxyID,
-					Dataplane:          dataplane,
-					TrafficPermissions: matchedPermissions,
-					TrafficRoutes:      routes,
-					OutboundSelectors:  destinations,
-					OutboundTargets:    outbound,
-					HealthChecks:       healthChecks,
-					CircuitBreakers:    circuitBreakers,
-					Logs:               matchedLogs,
-					TrafficTrace:       trafficTrace,
-					TracingBackend:     tracingBackend,
-					Metadata:           metadataTracker.Metadata(streamId),
-					FaultInjections:    faultInjection,
-					CLACache:           claCache,
-				}
-				err = reconciler.Reconcile(envoyCtx, &proxy)
-				if err != nil {
-					return err
-				}
-				prevHash = snapshotHash
-				return nil
-			},
-			OnError: func(err error) {
-				xdsGenerationsErrors.Inc()
-				log.Error(err, "OnTick() failed")
-			},
-			OnStop: func() {
-				proxyID := xds.FromResourceKey(key)
-				if err := reconciler.Clear(&proxyID); err != nil {
-					log.Error(err, "OnStop() failed")
-				}
-			},
-		}
-	}), nil
-}
-
-func DefaultDataplaneStatusTracker(rt core_runtime.Runtime) (DataplaneStatusTracker, error) {
-	tracker := NewDataplaneStatusTracker(rt, func(accessor SubscriptionStatusAccessor) DataplaneInsightSink {
-		return NewDataplaneInsightSink(
+func DefaultDataplaneStatusTracker(rt core_runtime.Runtime) (xds_callbacks.DataplaneStatusTracker, error) {
+	tracker := xds_callbacks.NewDataplaneStatusTracker(rt, func(accessor xds_callbacks.SubscriptionStatusAccessor) xds_callbacks.DataplaneInsightSink {
```

Comment:
I think it could be a single line `return` without `tracker` variable. And I believe we don't need the second return value `error`